src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -626,6 +626,14 @@ private void HandleFinished(bool success, ClientSideStatus receivedStatus)
             if (status.StatusCode != StatusCode.OK)
             {
                 streamingResponseCallFinishedTcs.SetException(new RpcException(status, receivedStatus.Trailers));
+                if (status.StatusCode == StatusCode.Cancelled)
+                {
+                    // Make sure the exception set to the Task is observed,
+                    // otherwise this can trigger "Unobserved exception" when the response stream
+                    // is not read until its end and the task created by the TCS is garbage collected.
+                    // See https://github.com/grpc/grpc/issues/17458
+                    var _ = streamingResponseCallFinishedTcs.Task.Exception;
```

Comment:
I think that wouldn't be useful as in most cases this exception actually will be observed in the user code (and we cannot know in advance if it will be).